### PR TITLE
feat(tracing) added support for https tracing endpoint

### LINKED ISSUE #2613: Support https endpoint for Tracing server.
Right now kuma only supports http endpoint for tracing server. This request is to support https endpoint too.

### PR DESCRIPTION
**Pull request by sudeeptoroy**
### Summary

added support for https tracing endpoints 

### Issues resolved

Fix #2613 

### Testing

- [ *] Unit tests

### Backwards compatibility

- [* ] `backport-to-stable`

### CODE REVIEW

File: pkg/xds/envoy/clusters/v3/dns_cluster_configurer_test.go [hunk 1]
```diff
@@ -12,12 +13,36 @@ import (
...
-		clusterName := "test:cluster"
-		address := "google.com"
-		port := uint32(80)
-		expected := `
+	type testCase struct {
+		clusterName string
+		address     string
+		port        int32
+		expected    string
+		isHttp      bool
```
Comment (lahabana) [inline pkg/xds/envoy/clusters/v3/dns_cluster_configurer_test.go:21]: Is this meant to be https ?

**Review by lahabana — CHANGES_REQUESTED** [review]
LGTM thank you for the contribution!

File: pkg/xds/envoy/clusters/v3/dns_cluster_configurer_test.go (hunk 1 above)
Comment (sudeeptoroy) [reply pkg/xds/envoy/clusters/v3/dns_cluster_configurer_test.go]: yes, you are right.. i will fix it.

**Review by lahabana — APPROVED** [review]
LGTM!

**Pull request merged**